### ISSUE TITLE
Vulkan: Assigning a uniform to a shader output crashes the editor.

### ISSUE THREAD

**Issue by wardensdev** [godotengine/godot#45913]
**Godot version:**

[ #45879](https://github.com/godotengine/godot/commit/ad293a82f1fc2e4baa35f54b9b8038df72d23737)

**OS/device including version:**

OS: Windows 10 - Version 10.0.19041 - Build 19041
GPU: RTX 2070S - Driver version 461.40

**Issue description:**

***This issue is not exclusive to VisualShaders, and seems to be a deeper problem***

The editor crashes when you connect a TextureUniform node to the Normal Map output in visual shaders.

These errors are shown in the console on launch of the editor:
```
ERROR: Condition "!shape->is_configured()" is true. Returning: Variant()
   at: shape_get_data (servers/physics_3d/physics_server_3d_sw.cpp:125)
ERROR: Condition "!d.has("cell_size")" is true.
   at: set_data (servers/physics_3d/shape_3d_sw.cpp:1688)
ERROR: Condition "!body" is true.
   at: body_attach_object_instance_id (servers/physics_3d/physics_server_3d_sw.cpp:614)
ERROR: Invalid ID.
   at: (servers/physics_3d/physics_server_3d_sw.cpp:1322)
```

This error is shown briefly before the editor crashes:

```
WARNING: A Thread object has been destroyed without wait_to_finish() having been called on it. Please do so to ensure correct cleanup of the thread.
     at: ~Thread (core/os/thread.cpp:111)
libc++abi: Pure virtual function called!
```

Apologies if this is a dup, I searched for similar issues the best I could. It seems shaders in general have had some issues the past couple of weeks, so this very well may be a broader problem than this specific circumstance.

**Steps to reproduce:**

- Open the reproduction project. 
- Navigate to: MeshInstance3D -> mesh -> material -> VisualShader
- Connect the TextureUniform to the Normal Map output in the shader graph.
- Click on a different node in the scene tree, and then click back to the MeshInstance3D.
- Editor will crash after a few seconds.

**Minimal reproduction project:**

[NormalMapIssue.zip](https://github.com/godotengine/godot/files/5969357/NormalMapIssue.zip)

**Comment by wardensdev** [comment]
I was digging around this issue a bit more today. When I re-opened the minimum reproduction project, I got this output in the console (no changes):

```
ERROR: Condition "!uniform_set" is true.
   at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:6454)
ERROR: Uniforms supplied for set (3):
Set: 3 Binding: 0 Type: UniformBuffer Length: 128
Set: 3 Binding: 1 Type: Texture Length: 1
Set: 3 Binding: 2 Type: Texture Length: 1
Set: 3 Binding: 3 Type: Texture Length: 1
are not the same format as required by the pipeline shader. Pipeline shader requires the following bindings:
Set: 0 Binding: 1 Type: Sampler Length: 12
Set: 0 Binding: 2 Type: Sampler Length: 1
Set: 0 Binding: 3 Type: StorageBuffer Length: 0
Set: 0 Binding: 4 Type: StorageBuffer Length: 0
Set: 0 Binding: 5 Type: StorageBuffer Length: 0
Set: 0 Binding: 6 Type: UniformBuffer Length: 4224
Set: 0 Binding: 7 Type: StorageBuffer Length: 0
Set: 0 Binding: 8 Type: StorageBuffer Length: 0
Set: 0 Binding: 9 Type: Texture Length: 1
Set: 0 Binding: 10 Type: Texture Length: 1
Set: 0 Binding: 11 Type: StorageBuffer Length: 0
Set: 0 Binding: 12 Type: StorageBuffer Length: 0
Set: 0 Binding: 13 Type: UniformBuffer Length: 368
Set: 1 Binding: 0 Type: UniformBuffer Length: 2672
Set: 1 Binding: 1 Type: StorageBuffer Length: 0
Set: 1 Binding: 2 Type: Texture Length: 1
Set: 1 Binding: 3 Type: Texture Length: 1
Set: 1 Binding: 4 Type: Texture Length: 1
Set: 1 Binding: 5 Type: Texture Length: 1
Set: 1 Binding: 6 Type: Texture Length: 8
Set: 1 Binding: 7 Type: Texture Length: 8
Set: 1 Binding: 8 Type: StorageBuffer Length: 0
Set: 1 Binding: 9 Type: Texture Length: 1
Set: 1 Binding: 10 Type: Texture Length: 1
Set: 1 Binding: 11 Type: Texture Length: 1
Set: 1 Binding: 12 Type: Texture Length: 1
Set: 1 Binding: 13 Type: Texture Length: 1
Set: 1 Binding: 14 Type: Texture Length: 1
Set: 1 Binding: 15 Type: Texture Length: 1
Set: 1 Binding: 16 Type: Texture Length: 1
Set: 1 Binding: 17 Type: UniformBuffer Length: 896
Set: 1 Binding: 18 Type: Texture Length: 1
Set: 2 Binding: 0 Type: StorageBuffer Length: 0
Set: 3 Binding: 0 Type: UniformBuffer Length: 16
Set: 3 Binding: 1 Type: Texture Length: 1
   at: (drivers/vulkan/rendering_device_vulkan.cpp:6612)
```

I started a new project with the same scene tree setup as the minimum reproduction project, but put a texture uniform in the vertex shader node graph. After the original crash denoted at the top of the issue, I re-opened it and got this shader code output, which I have saved in the .txt file below because it's incredibly long.

[ShaderErrorOutput.txt](https://github.com/godotengine/godot/files/5973975/ShaderErrorOutput.txt)

**Comment by wardensdev** [comment]
I've continued testing, this time with a written shader. It happens exactly the same as with the visual shader.

[UniformIssue.zip](https://github.com/godotengine/godot/files/5974685/UniformIssue.zip)

Just uncomment the code in the MeshInstance3D `mesh -> material -> shader`, change the uniform value in the editor, click on a different object in the scene tree, and click back to the MeshInstance3D. It will crash with the error output in my original

**Comment by Nanox19435** [comment]
Good news! This was resolved in 9ca0d66a3b51084eb528a85e133d9037b21bdd09.

**Comment by akien-mga** [comment]
Fixed by #46984.
